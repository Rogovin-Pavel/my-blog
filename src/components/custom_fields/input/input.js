import React, { Component } from 'react';

class Input extends Component {
  render() {
    let { type, 
      placeholder, 
      title,
      small_text, 
      label, 
      class_name, 
      onInputChange } = this.props;

    let input_label,
        input_small_text;

    type = type ? type : 'text';

    input_label = <label htmlFor={`input__${type}`}>{label}</label>;
    input_small_text = <small id={`input__${type}`} className="form-text text-muted">{small_text}</small>;
  
    return(
      <div className="input-form">
        { label.length ? input_label : null }

        <input  type={type}
                className={`${class_name} input-form__input input__${type}`}
                value={title}
                id={`input__${type}`}
                aria-describedby={`input__${type}`}
                placeholder={placeholder ? placeholder : 'введите данные'}
                onChange={onInputChange}/>

        { small_text.length ? input_small_text : null }
      </div>
    )
  }
}

export {
  Input
}