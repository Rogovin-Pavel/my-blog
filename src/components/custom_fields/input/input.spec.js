import React from 'react';
import renderer from 'react-test-renderer';

import { Input } from './input';
describe('Counter', () => {
  const type = 'text', 
        placeholder = 'insert your text', 
        title = 'title',
        small_text = 'small text', 
        label = 'label', 
        class_name = 'text input', 
        onInputChange = function() {};

  test('snapshot renders', () => {
    const component = renderer.create(<Input 
        type={type}
        placeholder={placeholder}
        title={title}
        small_text={small_text}
        label={label}
        class_name={class_name}
        onInputChange={onInputChange} 
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});