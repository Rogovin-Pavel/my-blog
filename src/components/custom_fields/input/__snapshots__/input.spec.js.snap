// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter snapshot renders 1`] = `
<div
  className="input-form"
>
  <label
    htmlFor="input__text"
  >
    label
  </label>
  <input
    aria-describedby="input__text"
    className="text input input-form__input input__text"
    id="input__text"
    onChange={[Function]}
    placeholder="insert your text"
    type="text"
    value="title"
  />
  <small
    className="form-text text-muted"
    id="input__text"
  >
    small text
  </small>
</div>
`;
